File: example/example-no-engine/src/main/java/io/holunda/camunda/bpm/data/example/noengine/Order.java
package io.holunda.camunda.bpm.data.example.noengine;

import java.time.Instant;
import java.util.List;

public class Order {

  private final String orderId;
  private final Instant createdOn;
  private final List<OrderPosition> positions;

  public Order(String orderId, Instant createdOn, List<OrderPosition> positions) {
    this.orderId = orderId;
    this.createdOn = createdOn;
    this.positions = positions;
  }

  public String getOrderId() {
    return orderId;
  }

  public Instant getCreatedOn() {
    return createdOn;
  }

  public List<OrderPosition> getPositions() {
    return positions;
  }

  @Override
  public String toString() {
    return "Order{" +
      "orderId='" + orderId + '\'' +
      ", createdOn=" + createdOn +
      ", positions=" + positions +
      '}';
  }
}

File: example/example-kotlin/src/main/kotlin/domain/Order.kt
package io.holunda.camunda.bpm.data.example.kotlin.domain

import java.util.*

/**
 * Order business entity.
 */
data class Order(
  /**
   * Order id.
   */
  val orderId: String,
  /**
   * Order create date.
   */
  val created: Date,
  /**
   * List of order positions.
   */
  val positions: List<OrderPosition> = listOf()
)
